programming/sed-article.md

```diff
@@ -2,88 +2,75 @@
 
 ## Overview
 
-`sed` is a Unix utility that transforms text.
-The name is short for "stream editor".
-Input comes from a stream of text and
-output is an edited version of that text.
+`sed` is a Unix text-processing utility.
+The name `sed` is short for "stream editor".
+Like other filters in the Unix pipe-and-filter paradigm, the input to `sed` is a stream of text and the output is an edited version of that text.
+`sed` has been branded a "power tool" and "Swiss Army Knife" and while most developers don't need a stream editor
+on a daily basis, `sed` is a great tool for your automated text manipulation toolbox.
 
-While most developers don't need a stream editor
-on a daily basis, `sed` is a great tool belt addition.
-
-Memorizing all the capabilities and syntax details
-of `sed` can be a daunting task.
-But after reading this article you will know enough
-to recognize when using `sed` is appropriate
-and can come back to it later for a refresher.
+After reading this article you will know enough
+to recognize when using `sed` is appropriate and will be familiar with the features of `sed` that are used most often in practice.
+When necessary, you can return to this article for a refresher on the more advanced features of `sed`.
 
 `sed` is based on the `ed` editor
 which was created by Ken Thompson in 1969.
 `sed` was created Lee McMahon of Bell Labs in 1973.
-Currently the most popular version is GNU sed.
+Currently, the most popular version is GNU sed.
 This is available on all major operating systems
 including Linux, macOS, and Windows.
 
-Alternatives to `sed` include the Unix utility `awk`
+Alternatives to `sed` include the Unix utility `awk`, `perl`,
```

Comment:
Why do you think I should specifically call out Perl since it is a general purpose programming language?

Reply:
When I think of text processing, my thought process goes sed -> awk -> perl/python -> something else.  Since perl was designed specifically for text processing, I think it is worth mentioning.